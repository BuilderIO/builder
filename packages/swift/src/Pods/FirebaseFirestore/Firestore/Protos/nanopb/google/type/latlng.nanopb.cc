/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* Automatically generated nanopb constant definitions */
/* Generated by nanopb-0.3.9.8 */

#include "latlng.nanopb.h"

#include "Firestore/core/src/nanopb/pretty_printing.h"

namespace firebase {
namespace firestore {

using nanopb::PrintEnumField;
using nanopb::PrintHeader;
using nanopb::PrintMessageField;
using nanopb::PrintPrimitiveField;
using nanopb::PrintTail;

/* @@protoc_insertion_point(includes) */
#if PB_PROTO_HEADER_VERSION != 30
#error Regenerate this file with the current version of nanopb generator.
#endif



const pb_field_t google_type_LatLng_fields[3] = {
    PB_FIELD(  1, DOUBLE  , SINGULAR, STATIC  , FIRST, google_type_LatLng, latitude, latitude, 0),
    PB_FIELD(  2, DOUBLE  , SINGULAR, STATIC  , OTHER, google_type_LatLng, longitude, latitude, 0),
    PB_LAST_FIELD
};


/* On some platforms (such as AVR), double is really float.
 * These are not directly supported by nanopb, but see example_avr_double.
 * To get rid of this error, remove any double fields from your .proto.
 */
PB_STATIC_ASSERT(sizeof(double) == 8, DOUBLE_MUST_BE_8_BYTES)

std::string google_type_LatLng::ToString(int indent) const {
    std::string header = PrintHeader(indent, "LatLng", this);
    std::string result;

    result += PrintPrimitiveField("latitude: ", latitude, indent + 1, false);
    result += PrintPrimitiveField("longitude: ", longitude, indent + 1, false);

    bool is_root = indent == 0;
    if (!result.empty() || is_root) {
      std::string tail = PrintTail(indent);
      return header + result + tail;
    } else {
      return "";
    }
}

}  // namespace firestore
}  // namespace firebase

/* @@protoc_insertion_point(eof) */
