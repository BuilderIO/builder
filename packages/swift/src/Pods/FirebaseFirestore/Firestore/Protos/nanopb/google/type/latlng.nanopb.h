/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* Automatically generated nanopb header */
/* Generated by nanopb-0.3.9.8 */

#ifndef PB_GOOGLE_TYPE_LATLNG_NANOPB_H_INCLUDED
#define PB_GOOGLE_TYPE_LATLNG_NANOPB_H_INCLUDED
#include <pb.h>

#include <string>

namespace firebase {
namespace firestore {

/* @@protoc_insertion_point(includes) */
#if PB_PROTO_HEADER_VERSION != 30
#error Regenerate this file with the current version of nanopb generator.
#endif


/* Struct definitions */
typedef struct _google_type_LatLng {
    double latitude;
    double longitude;

    std::string ToString(int indent = 0) const;
/* @@protoc_insertion_point(struct:google_type_LatLng) */
} google_type_LatLng;

/* Default values for struct fields */

/* Initializer values for message structs */
#define google_type_LatLng_init_default          {0, 0}
#define google_type_LatLng_init_zero             {0, 0}

/* Field tags (for use in manual encoding/decoding) */
#define google_type_LatLng_latitude_tag          1
#define google_type_LatLng_longitude_tag         2

/* Struct field encoding specification for nanopb */
extern const pb_field_t google_type_LatLng_fields[3];

/* Maximum encoded size of messages (where known) */
#define google_type_LatLng_size                  18

/* Message IDs (where set with "msgid" option) */
#ifdef PB_MSGID

#define LATLNG_MESSAGES \


#endif

}  // namespace firestore
}  // namespace firebase

/* @@protoc_insertion_point(eof) */

#endif
