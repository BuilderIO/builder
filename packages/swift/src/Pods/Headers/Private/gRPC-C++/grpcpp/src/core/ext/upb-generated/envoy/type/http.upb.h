../../../../../../../../../../gRPC-C++/src/core/ext/upb-generated/envoy/type/http.upb.h