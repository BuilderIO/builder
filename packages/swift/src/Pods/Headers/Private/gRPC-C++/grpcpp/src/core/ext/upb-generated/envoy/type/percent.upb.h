../../../../../../../../../../gRPC-C++/src/core/ext/upb-generated/envoy/type/percent.upb.h