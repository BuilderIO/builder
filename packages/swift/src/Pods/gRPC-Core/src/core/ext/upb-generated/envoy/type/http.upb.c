/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/http.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#include <stddef.h>
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "src/core/ext/upb-generated/envoy/type/http.upb.h"
#else
  #include  "envoy/type/http.upb.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

