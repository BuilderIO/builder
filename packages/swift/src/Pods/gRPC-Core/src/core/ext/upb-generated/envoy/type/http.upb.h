/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/http.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#ifndef ENVOY_TYPE_HTTP_PROTO_UPB_H_
#define ENVOY_TYPE_HTTP_PROTO_UPB_H_

#if COCOAPODS==1
  #include  "third_party/upb/upb/generated_util.h"
#else
  #include  "upb/generated_util.h"
#endif
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "third_party/upb/upb/decode.h"
#else
  #include  "upb/decode.h"
#endif
#if COCOAPODS==1
  #include  "third_party/upb/upb/encode.h"
#else
  #include  "upb/encode.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

#ifdef __cplusplus
extern "C" {
#endif

typedef enum {
  envoy_type_HTTP1 = 0,
  envoy_type_HTTP2 = 1,
  envoy_type_HTTP3 = 2
} envoy_type_CodecClientType;


#ifdef __cplusplus
}  /* extern "C" */
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

#endif  /* ENVOY_TYPE_HTTP_PROTO_UPB_H_ */
