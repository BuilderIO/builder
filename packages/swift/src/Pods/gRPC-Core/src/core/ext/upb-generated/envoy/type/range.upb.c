/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/range.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#include <stddef.h>
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "src/core/ext/upb-generated/envoy/type/range.upb.h"
#else
  #include  "envoy/type/range.upb.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

static const upb_msglayout_field envoy_type_Int64Range__fields[2] = {
  {1, UPB_SIZE(0, 0), 0, 0, 3, 1},
  {2, UPB_SIZE(8, 8), 0, 0, 3, 1},
};

const upb_msglayout envoy_type_Int64Range_msginit = {
  NULL,
  &envoy_type_Int64Range__fields[0],
  UPB_SIZE(16, 16), 2, false,
};

static const upb_msglayout_field envoy_type_Int32Range__fields[2] = {
  {1, UPB_SIZE(0, 0), 0, 0, 5, 1},
  {2, UPB_SIZE(4, 4), 0, 0, 5, 1},
};

const upb_msglayout envoy_type_Int32Range_msginit = {
  NULL,
  &envoy_type_Int32Range__fields[0],
  UPB_SIZE(8, 8), 2, false,
};

static const upb_msglayout_field envoy_type_DoubleRange__fields[2] = {
  {1, UPB_SIZE(0, 0), 0, 0, 1, 1},
  {2, UPB_SIZE(8, 8), 0, 0, 1, 1},
};

const upb_msglayout envoy_type_DoubleRange_msginit = {
  NULL,
  &envoy_type_DoubleRange__fields[0],
  UPB_SIZE(16, 16), 2, false,
};

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

