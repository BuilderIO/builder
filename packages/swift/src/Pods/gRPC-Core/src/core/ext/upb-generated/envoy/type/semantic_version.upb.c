/* This file was generated by upbc (the upb compiler) from the input
 * file:
 *
 *     envoy/type/semantic_version.proto
 *
 * Do not edit -- your changes will be discarded when the file is
 * regenerated. */

#include <stddef.h>
#if COCOAPODS==1
  #include  "third_party/upb/upb/msg.h"
#else
  #include  "upb/msg.h"
#endif
#if COCOAPODS==1
  #include  "src/core/ext/upb-generated/envoy/type/semantic_version.upb.h"
#else
  #include  "envoy/type/semantic_version.upb.h"
#endif

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_def.inc"
#else
  #include  "upb/port_def.inc"
#endif

static const upb_msglayout_field envoy_type_SemanticVersion__fields[3] = {
  {1, UPB_SIZE(0, 0), 0, 0, 13, 1},
  {2, UPB_SIZE(4, 4), 0, 0, 13, 1},
  {3, UPB_SIZE(8, 8), 0, 0, 13, 1},
};

const upb_msglayout envoy_type_SemanticVersion_msginit = {
  NULL,
  &envoy_type_SemanticVersion__fields[0],
  UPB_SIZE(12, 12), 3, false,
};

#if COCOAPODS==1
  #include  "third_party/upb/upb/port_undef.inc"
#else
  #include  "upb/port_undef.inc"
#endif

