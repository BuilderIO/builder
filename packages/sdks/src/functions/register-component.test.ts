import type { ComponentInfo } from '../types/components.js';
import {
  createRegisterComponentMessage,
  serializeComponentInfo,
} from './register-component.js';

describe('Component Registration and Serialization', () => {
  test('createRegisterComponentMessage creates correct message structure', () => {
    const mockComponentInfo: ComponentInfo = {
      name: 'TestComponent',
      inputs: [{ name: 'testInput', type: 'string' }],
      // Add other required fields as necessary
    };

    const result = createRegisterComponentMessage(mockComponentInfo);

    expect(result).toEqual({
      type: 'builder.registerComponent',
      data: expect.any(Object), // We'll test the data content separately
    });
  });

  test('serializeComponentInfo handles functions correctly', () => {
    const mockComponentInfo: ComponentInfo = {
      name: 'TestComponent',
      inputs: [{ name: 'testInput', type: 'string' }],
      hooks: {
        // eslint-disable-next-line
        onChange: function (value: string) {
          return value.toUpperCase();
        },
      },
      // Add other required fields as necessary
    };

    const result = serializeComponentInfo(mockComponentInfo);

    expect(result).toMatchSnapshot();
    expect(result.name).toBe('TestComponent');
    expect(result.inputs).toEqual([{ name: 'testInput', type: 'string' }]);
    expect(typeof result.hooks.onChange).toBe('string');
    expect(result.hooks.onChange).toContain('return value.toUpperCase();');
  });

  test('serializeComponentInfo handles arrow functions', () => {
    const mockComponentInfo: ComponentInfo = {
      name: 'ArrowComponent',
      inputs: [{ name: 'testInput', type: 'number' }],
      hooks: {
        onInit: (num: number) => num * 2,
      },
      // Add other required fields as necessary
    };

    const result = serializeComponentInfo(mockComponentInfo);

    expect(result).toMatchSnapshot();
    expect(typeof result.hooks.onInit).toBe('string');
    expect(result.hooks.onInit).toContain('num * 2');
  });

  test('serializeFn handles different function syntaxes', () => {
    const mockComponentInfo: ComponentInfo = {
      name: 'SyntaxTestComponent',
      inputs: [{ name: 'testInput', type: 'string' }],
      hooks: {
        func1: function namedFunction(x: number) {
          return x + 1;
        },
        func2: (x: number) => x * 2,
        func3(x: number) {
          return x - 1;
        },
      },
      // Add other required fields as necessary
    };

    const result = serializeComponentInfo(mockComponentInfo);

    expect(result).toMatchSnapshot();
  });
});
