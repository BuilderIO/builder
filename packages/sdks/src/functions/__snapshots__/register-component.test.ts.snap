// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component Registration and Serialization > serializeComponentInfo handles arrow functions 1`] = `
{
  "hooks": {
    "onInit": "return ((num) => num * 2).apply(this, arguments)",
  },
  "inputs": [
    {
      "name": "testInput",
      "type": "number",
    },
  ],
  "name": "ArrowComponent",
}
`;

exports[`Component Registration and Serialization > serializeComponentInfo handles functions correctly 1`] = `
{
  "hooks": {
    "onChange": "return (function(value) {
          return value.toUpperCase();
        }).apply(this, arguments)",
  },
  "inputs": [
    {
      "name": "testInput",
      "type": "string",
    },
  ],
  "name": "TestComponent",
}
`;

exports[`Component Registration and Serialization > serializeFn handles different function syntaxes 1`] = `
{
  "hooks": {
    "func1": "return (function namedFunction(x) {
          return x + 1;
        }).apply(this, arguments)",
    "func2": "return ((x) => x * 2).apply(this, arguments)",
    "func3": "return (function func3(x) {
          return x - 1;
        }).apply(this, arguments)",
  },
  "inputs": [
    {
      "name": "testInput",
      "type": "string",
    },
  ],
  "name": "SyntaxTestComponent",
}
`;
